kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluator.java

```diff
@@ -445,6 +459,27 @@ ColumnVector visitComparator(Predicate predicate) {
             return new DefaultBooleanVector(numRows, Optional.of(nullability), result);
         }
 
+        @Override
+        ColumnVector visitNullSafeComparator(Predicate predicate) {
+            PredicateChildrenEvalResult argResults = evalBinaryExpressionChildren(predicate);
+            int numRows = argResults.rowCount;
+            boolean[] result = new boolean[numRows];
+            int[] compareResult = compareNullSafe(argResults.leftResult, argResults.rightResult);
```

Comment:
we want to get away from this model to lazy compute model (#2541)

this can be rewritten as
```
// Create a comparator
// Basically this comparator is based on the data type.
Comparator comparator = ...

return new ColumnVector() {
   public getDataType() { return BooleanType.BOOLEAN;}

   public getSize() { argResults.leftResult.getSize(); }

   public isNullAt(int rowId) {
       boolean isLeftNull = argResults.leftResult.isNullAt(rowId);
       boolean isRightNull = argResults.rightResult.isNullAt(rowId);
       return (isLeftNull && !isRightNull) || (!isLeftNull && isRightNull);
   }

  public getBoolean(int rowId) {
      // check both are null. if yes return true.
      compartor.compare(leftResult, rightResult, rowId)
   }

}
```

Reply:
I am a bit confused on 
```
   public isNullAt(int rowId) {
       boolean isLeftNull = argResults.leftResult.isNullAt(rowId);
       boolean isRightNull = argResults.rightResult.isNullAt(rowId);
       return (isLeftNull && !isRightNull) || (!isLeftNull && isRightNull);
   }
```

Shouldn't `isNullAt` always return false based on https://spark.apache.org/docs/latest/sql-ref-null-semantics.html#comparison-operators-
because <=> can never return null?